front-end/src/ui-components/ScrollToTopWrapper.tsx

```diff
@@ -0,0 +1,70 @@
+// Copyright 2019-2020 @paritytech/polkassembly authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import styled from '@xstyled/styled-components';
+import React, { Component } from 'react';
+import { Icon } from 'semantic-ui-react';
+import Button from 'src/ui-components/Button';
+
+interface Props {
+	hasScrolled?: boolean
+}
+
+class ScrollingWrapper extends Component<{}, Props> {
+	constructor(props: any) {
+		super(props);
+		this.state = { hasScrolled: false };
+	}
+
+	componentDidMount() {
```

Comment:
We are using functional components. Can't we use this useEffect code to handle component did mount behaviour.
```js
  useEffect(() => {
    function handleScroll() {
      const scrollTop = window.scrollY;
      console.log(scrollTop);
    }

    window.addEventListener('scroll', handleScroll);

    return () => {
      window.removeEventListener('scroll', handleScroll);
    };
  }, []);
```

Reply:
And then set button visibility in `handleScroll`?

```typescript
const [buttonVisible, setButtonVisible] = useState(false);

	useEffect(() => {
		function handleScroll() {
			const scrollTop = window.scrollY;
			console.log(scrollTop);
			scrollTop > 300 ? setButtonVisible(true) : setButtonVisible(false);
		}

		window.addEventListener('scroll', handleScroll);

		return () => {
			window.removeEventListener('scroll', handleScroll);
		};
	}, []);

	const handleScrollToTop = () => {
		setButtonVisible(false);
		window.scrollTo({
			behavior: 'smooth',
			top: 0
		});
	};
```

This is a bit buggy, but maybe I'm doing something wrong here.

Reply:
what is the bug

Reply:
Button is flickering a bit after clicking. And seems to appear with a slight delay when initially loading the page and scrolling down

Reply:
Need to did deeper but for now class component is also good